sdk/audit/metadata_adding_interceptor_test.go

```diff
@@ -3,86 +3,162 @@ package audit
+	"net/http"
+	"net/http/httptest"
 	"testing"
 
+	"connectrpc.com/connect"
 	"github.com/google/uuid"
 	"github.com/opentdf/platform/protocol/go/kas"
+	"github.com/opentdf/platform/protocol/go/kas/kasconnect"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials/insecure"
 	"google.golang.org/grpc/metadata"
 	"google.golang.org/grpc/test/bufconn"
 )
 
-type FakeAccessServiceServer struct {
+type FakeAccessServiceServerConnect struct {
 	requestID uuid.UUID
 	requestIP string
 	actorID   string
-	kas.UnimplementedAccessServiceServer
+	kasconnect.UnimplementedAccessServiceHandler
 }
 
-func (f *FakeAccessServiceServer) PublicKey(ctx context.Context, _ *kas.PublicKeyRequest) (*kas.PublicKeyResponse, error) {
-	if md, ok := metadata.FromIncomingContext(ctx); ok {
-		requestIDFromMetadata := md.Get(string(RequestIDHeaderKey))
-		if len(requestIDFromMetadata) > 0 {
-			f.requestID, _ = uuid.Parse(requestIDFromMetadata[0])
-		}
+func (f *FakeAccessServiceServerConnect) PublicKey(_ context.Context, req *connect.Request[kas.PublicKeyRequest]) (*connect.Response[kas.PublicKeyResponse], error) {
+	requestIDFromHeader := req.Header().Get(string(RequestIDHeaderKey))
+	if requestIDFromHeader != "" {
+		f.requestID, _ = uuid.Parse(requestIDFromHeader)
+	}
 
-		requestIPFromMetadata := md.Get(string(RequestIPHeaderKey))
-		if len(requestIPFromMetadata) > 0 {
-			f.requestIP = requestIPFromMetadata[0]
-		}
+	requestIPFromHeader := req.Header().Get(string(RequestIPHeaderKey))
+	if requestIPFromHeader != "" {
+		f.requestIP = requestIPFromHeader
+	}
 
-		actorIDFromMetadata := md.Get(string(ActorIDHeaderKey))
-		if len(actorIDFromMetadata) > 0 {
-			f.actorID = actorIDFromMetadata[0]
-		}
+	actorIDFromHeader := req.Header().Get(string(ActorIDHeaderKey))
+	if actorIDFromHeader != "" {
+		f.actorID = actorIDFromHeader
 	}
-	return &kas.PublicKeyResponse{}, nil
+	return connect.NewResponse(&kas.PublicKeyResponse{}), nil
 }
 
 func TestAddingAuditMetadataToOutgoingRequest(t *testing.T) {
-	server := FakeAccessServiceServer{}
-	client, stop := runServer(&server)
-	defer stop()
+	serverConnect := FakeAccessServiceServerConnect{}
+	serverGrpc := FakeAccessServiceServer{}
+	clientConnect, stopC := runConnectServer(&serverConnect)
+	defer stopC()
+	clientGrpc, stopG := runServer(&serverGrpc)
+	defer stopG()
 
 	contextRequestID := uuid.New()
 	contextActorID := "actorID"
 	ctx := t.Context()
 	ctx = context.WithValue(ctx, RequestIDContextKey, contextRequestID)
 	ctx = context.WithValue(ctx, ActorIDContextKey, contextActorID)
 
-	_, err := client.PublicKey(ctx, &kas.PublicKeyRequest{})
+	_, err := clientConnect.PublicKey(ctx, connect.NewRequest(&kas.PublicKeyRequest{}))
 	if err != nil {
 		t.Fatalf("error making call: %v", err)
 	}
-
-	if server.requestID != contextRequestID {
-		t.Fatalf("request ID did not match: %v", server.requestID)
+	_, err = clientGrpc.PublicKey(ctx, &kas.PublicKeyRequest{})
+	if err != nil {
+		t.Fatalf("error making call: %v", err)
```

Comment:
feel free to upgrade to testify/assert (or require) to save a little bit of line count